src/tracing/instrumentation/mongodb.js

```diff
@@ -55,54 +62,74 @@ function instrument(mongodb) {
 
 
 function onStarted(event) {
-  if (!isActive ||
-      // In some cases, the operationId may not be defined on a MongoDB event!
-      !event.operationId) {
+  if (!isActive) {
+    return;
+  }
+
+  if (supportsOperationIds && !event.operationId) {
     return;
   }
 
-  var traceId = event.operationId.traceId;
-  var parentSpanId = event.operationId.parentSpanId;
   var parentSpan = cls.getCurrentSpan();
+  if (parentSpan && cls.isExitSpan(parentSpan)) {
+    return;
+  }
+
+  var traceId;
+  var parentSpanId;
+  // event.operationId is only present prior to mongodb package version 3.0.6, see above.
+  if (event.operationId) {
+    traceId = event.operationId.traceId;
+    parentSpanId = event.operationId.parentSpanId;
+    supportsOperationIds = true;
```

Comment:
As soon as we have seen one operationId attached to an event, we know that this is a version < 3.0.6.